src/coffee/run.coffee

```diff
@@ -54,6 +75,50 @@ zipFiles = (path, filename) ->
       buffer = zip.generate type: 'nodebuffer'
       fs.writeFileAsync "#{path}/#{filename}.zip", buffer, 'utf8'
 
+saveProductTypes = (result) ->
+  console.log 'About to write files...'
+  if _.isEmpty result.productTypes
+    Promise.reject new Error('We couldn\'t generate any file based on the given data. Please check your CSVs.')
+  else
+    Promise.map result.productTypes, (productType) ->
+      writeFileAsync productType, argv.target
+    .then ->
+      console.log "Generated #{_.size result.productTypes} files for normal product-types"
+      if argv.withRetailer
+        if _.isEmpty result.productTypesForRetailer
+          Promise.reject new Error('We couldn\'t generate any file for master<->retailer support based on the given data. Please check your CSVs.')
+        else
+          Promise.map result.productTypesForRetailer, (productType) ->
+            writeFileAsync productType, argv.target, 'retailer-product-type'
+          .then ->
+            console.log "Generated #{_.size result.productTypesForRetailer} files for retailer product-types"
+            Promise.resolve()
+      else
+        Promise.resolve()
+    .then ->
+      console.log "Finished generating files, checking results in target folder: #{argv.target}"
+      fs.readdirAsync argv.target
+      .then (files) ->
+        jsonFiles = _.filter files, (file) -> file.match(/\.json/)
+        if _.isEmpty jsonFiles
+          Promise.reject "No files were written in target folder #{argv.target}"
+        else
+          console.log "Found #{_.size jsonFiles} files in target folder #{argv.target}"
+          if argv.zip
+            console.log "Zipping files as #{argv.zipFileName} name"
+            zipFiles argv.target, argv.zipFileName
+          else
+            Promise.resolve()
+      .catch (e) ->
+        console.error "Oops, there was an error reading the files in target folder #{argv.target}: #{e.message}"
+        process.exit 1
+
+importSphereProductTypes = (data) ->
+  importer = new ProductTypeImporter
+  importer.init(argv)
+  .then ->
+    importer.import data
```

Comment:
@emmenko - please check update here 631bdd8 - I added singleton function which will wrap init together with constructor..

That init function returns promise because it uses `ProjectCredentialsConfig.create()` in module `sphere-node-utils` which does some async stuff (like reading config file I guess)